python_modules/libraries/dagster-duckdb/dagster_duckdb/io_manager.py

```diff
@@ -107,6 +110,26 @@ def duckdb_io_manager(init_context):
     return duckdb_io_manager
 
 
+def build_configurable_duckdb_io_manager(
+    type_handlers: Sequence[DbTypeHandler], default_load_type: Optional[Type] = None
+):
```

Comment:
I strongly recommend either accepting some copy paste here OR using a more OOP pattern rather than this class factory approach, which I think it both difficult to understand and difficult to type.


Option 1: OOP

Tell the use to write code like this and have a new base class that constructs the DBIOManager on the user's behalf

```
class UserDefinedDuckDbIOManager(CustomizedDuckDbIOManager):
   def type_handlers(self):
       # abstract method in base class
       return [UserDefinedTypeHandler()]
       
   def default_load_type(self):   
      return ...
```

Option 2: Copy paste

Just tell the user to copy and paste code and customize it to their needs.

```
class ExampleConfigurableIOManager(ConfigurableIOManagerFactory):
    database: str

    def create_io_manager(self, context) -> DbIOManager:
            return DbIOManager(
                db_client=DuckDbClient(),
                database=self.database,
                schema=self.schema_,
                type_handlers=... # put your type handlers here,
                default_load_type=# put your default load type here,
                io_manager_name="DuckDBIOManager",
        )
```

This makes DbIOManager a public API we have to support and feels like exposing internals. But at least it is obvious what is going to the typechecker and the user.